### PR TITLE
Improve/update Cloud Quickstart

### PR DESCRIPTION
**Pull request by petedejoy**
This PR realigns our quickstart positioning to better fit our new customer onboarding process being pushed in our new website.

### CODE REVIEW

File: cloud/README.md [hunk 1]
```diff
@@ -10,11 +10,9 @@ Astronomer Cloud is the easiest way for any organization to adopt [Apache Airflo
 
 Astronomer Cloud runs on a Kubernetes cluster hosted and managed by the Astronomer team. Our managed offering abstracts the everyday data engineer from the infrastructure management layer otherwise required to adopt, maintain and grow with Airflow.
 
-For a detailed breakdown of Astronomer Cloud, refer [here](/cloud/).
-
 ## Get Started
 
-Astronomer Cloud's [free 14-day Trial](/trial/) is the best way to get started. You're welcome to sign up at anytime, no credit card required.
+If you're interested in giving Astronomer Cloud a spin, you can start a free 14-day trial by [reaching out to us here](/get-astronomer?ref=docs). is the best way to get started. You're welcome to sign up at anytime, no credit card required.
```
Comment (paolaperaza) [inline cloud/README.md:15]:
```suggestion
If you're interested in trying out Astronomer Cloud, start a free 14-day trial by [reaching out to us here](/get-astronomer?ref=docs). You're welcome to sign up at anytime, no credit card required.
```

**Review by paolaperaza — CHANGES_REQUESTED** [review]
@petedejoy Spotted a typo/made one comment - lemme know once you fix and I'll pop back in here and merge! Good call to make these changes.

File: cloud/README.md (hunk 1 above)
Comment (paolaperaza) [reply cloud/README.md:15]: ("Giving Astronomer Cloud a spin" feels like too much like lingo/not accessible, so simplified it)

Comment (paolaperaza) [reply cloud/README.md]: Ok I fixed this :)

**Comment by paolaperaza** [comment]
@petedejoy After giving this original PR a quick review I realized that this doc (which is the first touchpoint for new users) was embarrassingly out of date given Airflow 2.0 and latest platform changes. There's still more that can be better, but I did a first pass at updating this. Feel free to review!

**Comment by petedejoy** [comment]
Great changes, going to merge this, thanks @paolaperaza!

**Pull request merged**